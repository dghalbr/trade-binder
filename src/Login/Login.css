.Login-center {
    margin: auto;
    display: block !important;
    width: 30%;
}