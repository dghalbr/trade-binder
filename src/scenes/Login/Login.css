.Login-center {
    margin: auto;
    display: block;
}