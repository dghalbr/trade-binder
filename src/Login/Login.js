import React, { Component } from 'react';
import './Login.css';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';

export default class Login extends Component {
  constructor(props) {
    super(props);
    this.login = this.login.bind(this);
  }

  /* DGH - 'Class properties are for data - methods are for functions. 
   *  If you're installing a function as a property that's anything more than a bound method, 
   *  you're doing something subpar..': https://github.com/airbnb/enzyme/issues/697
   *  Can't use arrow functions for these methods and plan on testing them.
   */
  login() {
    this.props.handleLogin(document.getElementById('username').value, document.getElementById('password').value);
    document.getElementById('username').value = null;
    document.getElementById('password').value = null;
  }

  render() {
    return (
      <div>
        <TextField  className="Login-center" 
                    type="text" 
                    id="username" 
                    floatingLabelText="Username"/>
        <TextField  className="Login-center" 
                    type="password" 
                    id="password"
                    floatingLabelText="Password" />
        <br />
        <RaisedButton   primary={true} 
                        className="Login-center" 
                        label="Login" 
                        id="login" 
                        onClick={this.login} />
      </div>
    );
  }
}
