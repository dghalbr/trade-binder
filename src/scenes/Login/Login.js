import React, { Component } from 'react';
import './Login.css';

export class Login extends Component {
  login = () => {
    this.props.handleLogin(document.getElementById('username').value, document.getElementById('password').value);
    document.getElementById('username').value = null;
    document.getElementById('password').value = null;
  };

  render() {
    return (
      <div>
        <input className='Login-center' type='text' id='username' />
        <input className='Login-center' type='password' id='password' />
        <button className='LoginButton Login-center' onClick={this.login}>Login</button>
      </div>
    );
  }
}
