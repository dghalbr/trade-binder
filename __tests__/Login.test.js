import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Login } from '../src/scenes/Login/Login';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { mount, configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactTestUtils from 'react-dom/test-utils'; 

configure({ adapter: new Adapter() });

describe('Component: Login', () => {
    it('should match its empty snapshot', () => {
        const tree = renderer.create(
            <Login />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should handle the click event', () => {
        // let handle = jest.fn();
        // let handle = () => {};reate(<Login handleLogin={ handle }/>)
        // const wrapper = shallow(<Login />);
        // wrapper.find(".LoginButton").simulate("click", { preventDefault() {} });

        const spy = jest.spyOn(Login.prototype, 'login');
const wrapper = mount(<Login  />);
wrapper.instance().methodName();
expect(spy).toHaveBeenCalled();

        // expect(handle).toHaveBeenCalled();
      });
});