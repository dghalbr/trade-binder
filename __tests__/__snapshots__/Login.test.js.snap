// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Login should match its empty snapshot 1`] = `
<div>
  <input
    className="Login-center"
    id="username"
    type="text"
  />
  <input
    className="Login-center"
    id="password"
    type="password"
  />
  <button
    className="LoginButton Login-center"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
